django/db/models/fields/related.py

```diff
@@ -1375,6 +1376,26 @@ def _check_relationship_model(self, from_model=None, **kwargs):
 
         return errors
 
+    def _check_table_uniqueness(self, **kwargs):
+        if self.db_table is None:
+            return []
```

Comment:
From a second read on the ticket it looks like we should always validate the table name even when not explicitly specified. You can remove this early return.